### PR TITLE
Upgrade payments web to java 21 and latest spring boot version

### PR DESCRIPTION
**Pull request by vshastri-ch**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/payments/controller/PaymentSummaryController.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import org.springframework.ui.Model;
 import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.*;
```
Comment (foman-ch) [inline src/main/java/uk/gov/companieshouse/web/payments/controller/PaymentSummaryController.java:8]: could remove the wildcard import :)

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -6,4 +6,10 @@ cdn.url=//${CDN_HOST}
...
 piwik.url=${PIWIK_URL}
 piwik.siteId=${PIWIK_SITE_ID}
-paymentPrivilegeKey=${CHS_PAYMENT_PRIVILEGE_API_KEY}
\ No newline at end of file
+paymentPrivilegeKey=${CHS_PAYMENT_PRIVILEGE_API_KEY}
+
+management.endpoints.enabled-by-default=false
+management.endpoints.web.exposure.include=health
+management.endpoints.web.base-path=/
+management.endpoints.web.path-mapping.health=payments-web/healthcheck
```
Comment (foman-ch) [inline src/main/resources/application.properties:14]: think you might need a forward slash at the start of this as well

File: src/test/java/uk/gov/companieshouse/web/payments/controller/PaymentSummaryControllerTests.java [hunk 3]
```diff
@@ -28,9 +28,7 @@
 import static org.mockito.Mockito.when;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/payments/controller/PaymentSummaryControllerTests.java:31]: wildcard import

**Review by foman-ch — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (foman-ch) [reply src/main/resources/application.properties:14]: think you can ignore this actually, seems like some services include and some don't

**Review by foman-ch — APPROVED** [review]

**Review by dhawes-ch — APPROVED** [review]

**Pull request merged**